### PR TITLE
Client server flow

### PR DESCRIPTION
**Pull request by mKiddeT**
Client is able to send video directly to the sanic backend api, and gets a response with the extracted landsmarks from mediapipe.
This is for the sake of the prototype. The finished product should be Client->Nextjs->Sanic, and should not circumvent Nextjs

### CODE REVIEW

File: nextjs/pages/components/WebcamCapture.tsx [hunk 1]
```diff
@@ -38,12 +39,22 @@ export default function WebcamCapture() {
...
+    
         const url = URL.createObjectURL(blob);
         const a = document.createElement("a");
         document.body.appendChild(a);
-        a.href = url;
-        a.download = "signvid.webm";
-        a.click();
+        //a.href = url;
+        //a.download = "signvid.webm";
+        //a.click();
```
Comment (CasperKjaerhus) [inline nextjs/pages/components/WebcamCapture.tsx]:
```suggestion
```

File: nextjs/pages/components/WebcamCapture.tsx [hunk 2]
```diff
@@ -1,5 +1,6 @@
 import Webcam from "react-webcam";
 import React from "react";
+import { json } from "stream/consumers";
```
Comment (CasperKjaerhus) [inline nextjs/pages/components/WebcamCapture.tsx]:
```suggestion
```

File: nextjs/pages/components/WebcamCapture.tsx [hunk 3]
```diff
@@ -38,12 +39,22 @@ export default function WebcamCapture() {
         const blob = new Blob(recordedChunks, {
             type: "video/webm"
         });
+
+        // Send to upload Python server  
+        var fd = new FormData();
```
Comment (CasperKjaerhus) [inline nextjs/pages/components/WebcamCapture.tsx]:
```suggestion
        const fd = new FormData();
```

File: nextjs/pages/index.tsx [hunk 4]
```diff
@@ -7,15 +7,18 @@ export default function LearningPage() {
...
+        <div className="self-center">
+          { showWebcam ? null : <VideoPlayer /> }
+          { showWebcam ? <WebcamCapture /> : null }
+        </div>
+      
+        <div className="self-center" >
+          {/* buttons */}
+          { showWebcam ? null : <button onClick={() => setShowWebcam(true)} className="bg-blue-500 hover:bg-blue-700 text-white font-bold py-2 px-4 rounded">Record</button> }
+          { showWebcam ? <button onClick={() => setShowWebcam(false)} className="bg-blue-500 hover:bg-blue-700 text-white font-bold py-2 px-4 rounded">Back to vid</button> : null }
+        </div>
```
Comment (CasperKjaerhus) [inline nextjs/pages/index.tsx:21]: based css

**Review by CasperKjaerhus — APPROVED** [review]

File: nextjs/pages/components/WebcamCapture.tsx [hunk 5]
```diff
@@ -38,12 +38,19 @@ export default function WebcamCapture() {
...
+        fd.append("fname", "video.webm")
+        fd.append("video", blob);
+        fetch("http://localhost:8080/api/hands", {
+            method: "POST",
+            body: fd,
+        });
+    
         const url = URL.createObjectURL(blob);
         const a = document.createElement("a");
         document.body.appendChild(a);
```
Comment (Deaxz) [inline nextjs/pages/components/WebcamCapture.tsx]: Slet hele funktionalitet i stedet for delmængde
```suggestion
```

**Review by Deaxz — CHANGES_REQUESTED** [review]

**Review by Deaxz — APPROVED** [review]

**Pull request merged**